plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -157,29 +157,34 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 		checkDuplicated(message, model as Model, newHashSet)	
 	}
 	
-	def boolean checkDuplicated(Message message, EObject model, HashSet<String> names) {
+	def boolean checkDuplicated(EObject elementToCheck, EObject model, HashSet<String> names) {

```

Comment:
Add a separate function to check duplicated combined fragments.
Do not edit in the same place as for messages, let this checkDuplicated unchanged.